src/data-interfaces/HoloFuelDnaInterface.js

```diff
@@ -133,7 +143,30 @@ function presentPendingOffer (transaction, annuled = false) {
   const counterpartyId = annuled ? event[2].Promise.tx.to : provenance[0]
   const { amount, notes, fee } = event[2].Promise.tx
   const isPayingARequest = !!event[2].Promise.request
-  return presentOffer({ origin, event: event[2], stateDirection, status, type, eventTimestamp, counterpartyId, amount, notes, fees: fee, isPayingARequest })
+  const inProcess = isEmpty(invoicedOffers) ? false : findEvent()
+  return presentOffer({ origin, event: event[2], stateDirection, status, type, eventTimestamp, counterpartyId, amount, notes, fees: fee, isPayingARequest, inProcess })
+}
+
+let counter = 0
+async function getListPending (params) {
+  const { requests, promises, declined } = await createZomeCall('transactions/list_pending')(params)
+  counter++
+  if (counter === 8) {
+    counter = 0
+    promises.forEach(p => {
+      if (!isEmpty(p[1])) {
+            findEvent(p[0],p[1])
+      }
+    })
+  }
```

Comment:
This code has side effects. The DNAInterfaces should as much as possible just be pure data transformations, and not have side effects. I would move this logic into the UI

Reply:
Let's talk about this more.